### ISSUE TITLE
XGB n_estimators of 500

### ISSUE THREAD

**Issue by ronmexico2718** [rhiever/tpot#272]
Is there any particular reason why n_estimators is set to 500 by default?  Seems like a fairly significant limitation.  Frequently I've experience significant performance gains by setting (outside of tpot) the learning rate to a small value.  However, with a small learning rate you'll want to use a larger value for n_estimators.

**Comment by rhiever** [comment]
Experimentally, we've found that (generally) `n_estimators=500` works well enough to get most of the gains that tree-based ensemble methods will achieve. Thus we fix `n_estimators` at a reasonably high value (considering both accuracy and time to build the model) and allow the learning rate to adapt around that. If you have evidence that this practice could be wrong, please let us know!

**Comment by rhiever** [comment]
Please let us know if you have any additional questions/comments about this topic. Happy to reopen the issue and continue the conversation.

**Comment by rspadim** [comment]
hi, it's relevant... i tested an xgboost at kaggle with 10 vs 100 n_estimators and the diference was from 100 position to 10 position at kaggle rank

could some parameters be "optional", for example...
if we use TPOTRegressor( n_jobs = -1, n_estimators= 100)
could the TPOTRegressor accept it and when some model (xgboost for example) know this parameter just use it? i see many tree using n_jobs and n_estimators, but some linear regressors no

maybe optional parameters could be interesting... or fixed, no problem, the point is expose more parameters instead of fixed values

could we reopen this issue?
if possible (i'm new to python) one example (commit) adding a new parameter to TPOTRegressor + xgboost parameter, could help allot increasing the complexity/number of parameters that we could use with TPOT

**Comment by rhiever** [comment]
In the upcoming 0.7 release, we'll be including the option to fully customize what TPOT optimizes over (estimators, selectors, parameters, parameter ranges). I think that will help with this situation.
